handler/handler.go

```diff
@@ -27,6 +28,7 @@ func CreateNatsQueue(address string, port int) (*NatsQueue, error) {
 	clusterID := "faas-cluster"
 
 	nc, err := stan.Connect(clusterID, clientID, stan.NatsURL(natsURL))
+	nc.NatsConn().SetReconnectHandler(queue1.reconnectClient(clientID, clusterID, natsURL))
```

Comment:
As it is, you should check for err because if there was an error, `nc` would be `nil` and you would get a panic here.

But this is a moot point. You should be creating the NATS connection first, configured with the reconnect handler. Then pass the NATS connection to the `stan.Connect()` call. Otherwise, you still potentially miss a window between the `stan.Connect()` call and the `nc.NatsConn().SetReconnectHandler()` where the connection may have disconnect/reconnect. It will also be easier on reconnect to simply recreate the stan connection (no need to close the connection for which you just got a notification that you were reconnected).

You would need to store the NATS connection (and close it after closing the stan connection) in your NatsQueue structure. By the way, since you would need to store both, I would recommend `sc` for stan connection and `nc` for nats connection. Right now, `nc` is used for the stan connection.